tests/CppUTest/CommandLineArgumentsTest.cpp

```diff
@@ -102,12 +111,74 @@ TEST(CommandLineArguments, repeatSetDifferentParameter)
     LONGS_EQUAL(4, args->getRepeatCount());
 }
 
-TEST(CommandLineArguments, repeatSetDefaultsToTwo)
+TEST(CommandLineArguments, repeatSetDefaultsToTwoAndShuffleDisabled)
 {
     int argc = 2;
     const char* argv[] = { "tests.exe", "-r" };
     CHECK(newArgumentParser(argc, argv));
     LONGS_EQUAL(2, args->getRepeatCount());
+    CHECK_EQUAL(args->getShuffle(), 0);
+}
+
+TEST(CommandLineArguments, shuffleEnabled)
+{
+    int argc = 2;
+    const char* argv[] = { "tests.exe", "-s" };
+    CHECK(newArgumentParser(argc, argv));
+    CHECK_EQUAL(args->getShuffle(), 1);
+}
+
+TEST(CommandLineArguments, shuffleEnabledSpecificSeedCase1)
+{
+    int argc = 2;
+    const char* argv[] = { "tests.exe", "-s999"};
+    CHECK(newArgumentParser(argc, argv));
+    CHECK_EQUAL(args->getShuffle(), 999);
+}
+
+TEST(CommandLineArguments, shuffleEnabledSpecificSeedCase2)
+{
+    int argc = 2;
+    const char* argv[] = { "tests.exe", "-s 888"};
+    CHECK(newArgumentParser(argc, argv));
+    CHECK_EQUAL(args->getShuffle(), 888);
+}
+
+TEST(CommandLineArguments, shuffleEnabledSpecificSeedCase3)
+{
+    int argc = 3;
+    const char* argv[] = { "tests.exe", "-s", "777"};
+    CHECK(newArgumentParser(argc, argv));
+    CHECK_EQUAL(args->getShuffle(), 777);
+}
+
+TEST(CommandLineArguments, shuffleBeforeDoesNotDisturbOtherSwitchCase1)
+{
+    const char* argv[] = { "tests.exe", "-s", "-sg", "group" };
+    shuffleDisturbTest(4, argv, 1);
```

Comment:
These tests don't test anything...

Reply:
In fact they do, although I trimmed the number of tests. `CommandLineArguments::parse` parses command line switches in order - if one parser misbehaves, it can break the following command line switches.

E.g., comment out `if (wasAbleToParseDigit)` in `CommandLineArguments::SetShuffle` --> switches following `-s` will not work, and `shuffleBeforeDoesNotDisturbOtherSwitch` unit test will catch.